### PR TITLE
Updated HistoricalPriceFeedV2Wrapper to TimeSeriesFeedWrapper, change…

### PR DESCRIPTION
**Pull request by bweick**
…d interface to match new variable name. Updated OracleAPI to use TimeSeriesFeedWrapper, updated getRollingHistoricalFeedNextAvailableUpdateAsync to getRollingHistoricalFeedNextEarliestUpdateAsync. Added helper functions for new TimeSeriesFeed system. Bumped to use new strategies package.

### CODE REVIEW

File: src/api/OracleAPI.ts [hunk 1]
```diff
@@ -64,13 +64,13 @@ export class OracleAPI {
...
-   * Returns the Unix timestamp of when price can feed can next be updated
+   * Returns the Unix timestamp of earliest time the TimeSeriesFeed can be updated
    *
-   * @param  historicalPriceFeed    Address of the HistoricalPriceFeedV2 contract to poll
+   * @param  timeSeriesFeed         Address of the TimeSeriesFeed contract to poll
    * @return                        Timestamp of when the price feed can be updated next
    */
-  public async getRollingHistoricalFeedNextAvailableUpdateAsync(historicalPriceFeed: Address): Promise<BigNumber> {
-    return await this.historicalPriceFeedV2Wrapper.nextAvailableUpdate(historicalPriceFeed);
+  public async getRollingHistoricalFeedNextEarliestUpdateAsync(timeSeriesFeed: Address): Promise<BigNumber> {
```
Comment (felix2feng) [inline src/api/OracleAPI.ts]: I get that we're trying to keep the interface the same. But would it be worth it to rename to `getTimeSeriesFeedNextEarliestUpdateAsync`?

File: src/wrappers/strategies/StrategyContractWrapper.ts [hunk 2]
```diff
@@ -74,28 +74,28 @@ export class StrategyContractWrapper {
   }
 
   /**
-   * Load a HistoricalPriceFeedV2 contract
+   * Load a TimeSeriesFeed contract
    *
-   * @param  historicalPriceFeed          Address of the HistoricalPriceFeedV2 contract
+   * @param  historicalPriceFeed          Address of the TimeSeriesFeed contract
```
Comment (felix2feng) [inline src/wrappers/strategies/StrategyContractWrapper.ts]: `timeSeriesFeed`

File: src/wrappers/strategies/TimeSeriesFeedWrapper.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ import { Address } from '../../types/common';
  * The HistoricalPriceFeedV2Wrapper handles interactions with Set's price feed for moving averages
```
Comment (felix2feng) [inline src/wrappers/strategies/TimeSeriesFeedWrapper.ts]: Update me and Line 26

File: test/helpers/oracleHelpers.ts [hunk 4]
```diff
@@ -90,32 +92,55 @@ export const deployHistoricalPriceFeedAsync = async(
...
-  seededValues: BigNumber[] = [],
+  seededValues: BigNumber[] = [new BigNumber(200 * 10 ** 18)],
   from: Address = TX_DEFAULTS.from,
-): Promise<HistoricalPriceFeedV2Contract> => {
-  const truffleHistoricalPriceFeedContract = contract(HistoricalPriceFeedV2);
-  truffleHistoricalPriceFeedContract.setProvider(web3.currentProvider);
-  truffleHistoricalPriceFeedContract.setNetwork(50);
-  truffleHistoricalPriceFeedContract.defaults(TX_DEFAULTS);
+): Promise<TimeSeriesFeedContract> => {
+  const truffleTimeSeriesFeedContract = contract(TimeSeriesFeed);
```
Comment (felix2feng) [inline test/helpers/oracleHelpers.ts]: @asoong At some point, we should create a helper method or class that consolidates these settings into a single function vs. duplication 4 lines per contract and refactor everything.

File: src/api/OracleAPI.ts (hunk 1 above)
Comment (bweick) [reply src/api/OracleAPI.ts]: The interface is already changed so I have no problem with that

File: src/api/OracleAPI.ts [hunk 5]
```diff
@@ -64,13 +64,13 @@ export class OracleAPI {
...
 
   /**
-   * Returns the Unix timestamp of when price can feed can next be updated
+   * Returns the Unix timestamp of earliest time the TimeSeriesFeed can be updated
    *
-   * @param  historicalPriceFeed    Address of the HistoricalPriceFeedV2 contract to poll
+   * @param  timeSeriesFeed         Address of the TimeSeriesFeed contract to poll
    * @return                        Timestamp of when the price feed can be updated next
    */
-  public async getRollingHistoricalFeedNextAvailableUpdateAsync(historicalPriceFeed: Address): Promise<BigNumber> {
```
Comment (felix2feng) [inline src/api/OracleAPI.ts]: I wonder how we can improve the interface of the functions in this API.

File: src/api/OracleAPI.ts [hunk 6]
```diff
@@ -48,59 +45,25 @@ export class OracleAPI {
    *                      with the Ethereum network
    */
   constructor(web3: Web3) {
-    this.historicalPriceFeedWrapper = new HistoricalPriceFeedWrapper(web3);
-    this.historicalPriceFeedV2Wrapper = new HistoricalPriceFeedV2Wrapper(web3);
     this.movingAverageOracleWrapper = new MovingAverageOracleWrapper(web3);
+    this.priceFeed = new PriceFeedWrapper(web3);
```
Comment (asoong) [inline src/api/OracleAPI.ts]: No point being coy here, might as well rename it to medianizer

**Pull request merged**